includes/blocks/class-sensei-course-blocks.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * File containing the class Sensei_Course_Blocks.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Sensei_Course_Blocks
+ */
+class Sensei_Course_Blocks {
+	/**
+	 * Course outline block.
+	 *
+	 * @var Sensei_Course_Outline_Block
+	 */
+	public $outline;
```

Comment:
I'm not a fan of public properties but it might be too much work to change it at this point.

Reply:
This is about the same as the current architecture of Sensei, with `Sensei()->thing` being a widely used way to access instances. It's also easy to stub when testing. This ends up like `Sensei()->blocks->course->outline`. 

Maybe an alternative can be just creating singletons for everything, like `Sensei_Course_Outline_Block::instance()`, but that adds a lot of boilerplate to every class.

Could also rework the block classes to create a new instance for every render, like `new Sensei_Course_Lesson_Block( $attributes, $course_id )`, each class implementing a `Sensei_Block` interface with a `render()` function. 

Also as a lighter alternative for dependency injection, we could make a factory that creates block classes by block name, like `Sensei_Blocks::create( 'sensei-lms/course-lesson' )`.

Probably in a future PR :)